pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/client/AssumeRoleClientProvider.scala

```diff
@@ -0,0 +1,46 @@
+package uk.ac.wellcome.platform.transformer.mets.client
+
+import java.util.concurrent.atomic.AtomicReference
+
+import akka.actor.ActorSystem
+import com.amazonaws.auth.BasicSessionCredentials
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService
+import com.amazonaws.services.securitytoken.model.AssumeRoleRequest
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.util.Try
+
+class AssumeRoleClientProvider[T](
+  stsClient: AWSSecurityTokenService,
+  roleArn: String,
+  interval: FiniteDuration = 30 seconds)(clientFactory: ClientFactory[T])(
+  implicit actorSystem: ActorSystem,
+  ec: ExecutionContext) {
+  val client = new AtomicReference[T]
+
+  actorSystem.scheduler.schedule(0 milliseconds, interval)(refreshClient())
+
+  def getClient: Either[Throwable, T] = {
+    Option(client.get()) match {
```

Comment:
Does this throw if the client has expired?

Reply:
it would throw a permission error I assume... 
Unfortunately, moto does not allow to test this. I can't find a way to recreate conditions like this (or, what happens if you try to assume a role that you don't have permissions to assume etc)

Reply:
Is it possible to override the client class, with the method in question throwing an error? Or more complicated than that?

Reply:
Actually, no, that wouldn't throw any error. It returns `null` if the `AtomicReference` is not initialised yet (which reminds me `client` should be private, I'll fix that). But because it's wrapped in `Option` it will always be `None` or `Some`.
If the client has expired, it would only show when we actually try to use it I think, so in `TemporaryCredentialsStore`